src/main/java/org/traccar/protocol/GatorProtocolDecoder.java

```diff
@@ -90,7 +97,56 @@ protected Object decode(
 
         sendResponse(channel, remoteAddress, type, buf.getByte(buf.writerIndex() - 2));
 
-        if (type == MSG_POSITION_DATA || type == MSG_ROLLCALL_RESPONSE
+        if (type == MSG_RFID_DATA) {
+            Position position = new Position(getProtocolName());
+
+            DeviceSession deviceSession = getDeviceSession(channel, remoteAddress, "1" + id, id);
+            if (deviceSession == null) {
+                return null;
+            }
+            position.setDeviceId(deviceSession.getDeviceId());
+
+            int dataType = buf.readUnsignedByte();
+
+            // RFID Data Type - 0x02
+            if (dataType != 0x02) {
+                return null;
+            }
+
+            StringBuilder rfidData = new StringBuilder();
+            int rfidDataEndIndex = buf.indexOf(buf.readerIndex(), buf.writerIndex(), (byte) 0x0D);
+
+            if (rfidDataEndIndex != -1) {
+                int length = rfidDataEndIndex - buf.readerIndex();
+                rfidData.append(buf.toString(buf.readerIndex(), length, Charset.defaultCharset()));
+
+                buf.skipBytes(length + 1);
+            } else {
+                return null;
+            }
+
+            position.set(Position.KEY_DRIVER_UNIQUE_ID, rfidData.toString());
```

Comment:
I previously asked to simplify this, but it still seems overcomplicated. Why do you need a `StringBuilder`?